src/dds/rs-dds-device-proxy.cpp

```diff
@@ -394,6 +394,18 @@ dds_device_proxy::dds_device_proxy( std::shared_ptr< const device_info > const &
         !strcmp(get_info(RS2_CAMERA_INFO_PRODUCT_LINE).c_str(), "D500"))
         set_auto_calibration_capability(std::make_shared<d500_auto_calibrated>(
             std::make_shared<d500_debug_protocol_calibration_engine>(this)));
+            
+    _calibration_changed_subscription = _dds_dev->on_calibration_changed(
+        [this]( std::shared_ptr< const realdds::dds_stream > const & stream )
+        {
+            auto it = _stream_name_to_profiles.find( stream->name() );
+            if( it == _stream_name_to_profiles.end() )
+                throw std::runtime_error( rsutils::string::from() << "no such stream?!" );
+
+            auto video_stream = std::dynamic_pointer_cast< const realdds::dds_video_stream >( stream );
```

Comment:
On failure to cast, should try as motion intrinsics and throw if not one of the two.

Reply:
Well, right now motion isn't supported for calibration changed.
But OK, I'll throw an error.